### PR TITLE
Adds inline documentation for upgradeability-lib contracts

### PR DESCRIPTION
**Pull request by fiiiu**
- [x] versioning
- [x] management

Partially addressing zeppelinos/labs#44.

### CODE REVIEW

File: contracts/application/management/BaseAppManager.sol [hunk 1]
```diff
@@ -5,40 +5,88 @@ import "../../upgradeability/OwnedUpgradeabilityProxy.sol";
...
+   * @dev Constructor function
+   * @param _factory Proxy factory
+   */
   function BaseAppManager(UpgradeabilityProxyFactory _factory) public {
     require(address(_factory) != address(0));
     factory = _factory;
   }
 
+  /**
+   * @dev Prototype of function for fetching the manager's contract provider
```
Comment (facuspagnuolo) [inline contracts/application/management/BaseAppManager.sol]: Prototype function

File: contracts/application/management/BaseAppManager.sol [hunk 2]
```diff
@@ -5,40 +5,88 @@ import "../../upgradeability/OwnedUpgradeabilityProxy.sol";
...
+   * @return Address of the new proxy
+   */
   function create(string contractName) public returns (OwnedUpgradeabilityProxy) {
     address implementation = getImplementationOrRevert(contractName);
     return factory.createProxy(this, implementation);
   }
 
-  function createAndCall(string contractName, bytes data) payable public returns (OwnedUpgradeabilityProxy) {
+  /**
+   * @dev Creates a new proxy for the given contract and forwards it the function call packed in data
```
Comment (facuspagnuolo) [inline contracts/application/management/BaseAppManager.sol]: and forwards a function call to it

File: contracts/application/management/BaseAppManager.sol [hunk 3]
```diff
@@ -5,40 +5,88 @@ import "../../upgradeability/OwnedUpgradeabilityProxy.sol";
...
+   * @dev Useful for initializing the proxied contract
+   * @param contractName Name of the contract for which a proxy is desired
+   * @param data Data to be sent as msg.data in the forwarded function call to the proxy, packing the methodId and parameters
```
Comment (facuspagnuolo) [inline contracts/application/management/BaseAppManager.sol]: I would say sth like `represents the msg.data to bet sent in the low level call. This parameter should include the signature of the function to called in the implementation with its corresponding payload` or so

File: contracts/application/management/BaseAppManager.sol [hunk 4]
```diff
@@ -5,40 +5,88 @@ import "../../upgradeability/OwnedUpgradeabilityProxy.sol";
...
   function upgradeTo(OwnedUpgradeabilityProxy proxy, string contractName) public onlyOwner {
     address implementation = getImplementationOrRevert(contractName);
     proxy.upgradeTo(implementation);
   }
 
+  /**
+   * @dev Upgrades a proxy to a new implementation of a contract and forwards it the function call packed in data
+   * @param proxy Proxy to be upgraded
+   * @param contractName Name of the contract with a new implmentation
+   * @param data Data to be sent as msg.data in the forwarded function call to the proxy, packing the methodId and parameters
```
Comment (facuspagnuolo) [inline contracts/application/management/BaseAppManager.sol]: same here

File: contracts/upgradeability/UpgradeabilityProxyFactory.sol [hunk 5]
```diff
@@ -47,9 +47,9 @@ contract UpgradeabilityProxyFactory {
   }
 
   /**
-  * @dev Internal function to create an upgradeable proxy
-  * @return address of the new proxy created
-  */
+   * @dev Internal function to create an upgradeable proxy
+   * @return address of the new proxy created
+   */
```
Comment (facuspagnuolo) [inline contracts/upgradeability/UpgradeabilityProxyFactory.sol]: there are many indentation changes, please confirm they were intended just in case...

Comment (fiiiu) [reply contracts/upgradeability/UpgradeabilityProxyFactory.sol]: yep, all checked to be homogeneous across contracts

**Pull request merged**